libcudacxx/include/cuda/__shuffle/warp_shuffle.h

```diff
@@ -0,0 +1,151 @@
+
+#ifndef _CUDA_FUNCTIONAL_SHUFFLE_SAFETY_H
+#define _CUDA_FUNCTIONAL_SHUFFLE_SAFETY_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/type_traits>
+#include <cuda/std/bit>
+#include <cuda/std/memory>
+#include <cuda/std/__algorithm/clamp.h>
+#include <cuda/__ptx/instructions/generated/get_sreg.h>
+#include <cuda/std/__cmath/nvfp16.h>
+#include <cuda/std/__cmath/nvbf16.h>
+#include <cuda/__cmath/ceil_div.h>
+
+
+#define _CCCL_HAS_CUDA_COMPILER 1 //fix for now -- to be deleted later
+
+#if _CCCL_HAS_CUDA_COMPILER
+_LIBCUDACXX_BEGIN_NAMESPACE_CUDA
+    template <typename T>
+    constexpr bool is_supported_type = cuda::std::is_trivially_copyable<T>::value;
```

Comment:
Needs to be named something like `__can_warp_shuffle`. Also this also needs to 
* exclude types smaller than int
* allow extended floating point types like `__half` and `__nv_bfloat16`